dora/core/common/src/main/java/alluxio/wire/WorkerIdentity.java

```diff
@@ -0,0 +1,363 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import com.google.common.primitives.Longs;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.UnsafeByteOperations;
+import org.apache.commons.codec.binary.Hex;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.UUID;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * An identifier that uniquely identifies a worker.
+ * The associated utility class {@link Parsers} provides means of converting between a GRPC
+ * representation and an instance of this class.
+ */
+@Immutable
+public final class WorkerIdentity {
+  private final byte[] mId;
+  private final int mVersion;
+  private final int mHashcode;
+
+  WorkerIdentity(byte[] id, int version) {
+    mId = id;
+    mVersion = version;
+    mHashcode = Arrays.hashCode(mId) * 31 + mVersion;
```

Comment:
if mId is already a long-representing byte[] wouldn't the additional 31 based polynomial with int type from this already be overflowed? and what about support even longer byte[]? i'm assuming its ok to overflow but just curious of how much of possibility of clash would happen if overflow happens, and how crucial would that be

Reply:
I borrowed this piece of code from `java.lang.String#hashCode()`. Collision is inevitable for a hash code. If two different objects have the same hash code, then the user needs to further compare them by `equals`. It's bad for performance, but wouldn't cause incorrect behavior.